drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -144,92 +221,200 @@ local function assign_child_profile(device, child_ep)
+      else -- device only supports momentary switch, no release events
+        device.log.debug("configuring for press event only")
+        supportedButtonValues_event = capabilities.button.supportedButtonValues({"pushed"}, {visibility = {displayed = false}})
+        set_field_for_endpoint(device, INITIAL_PRESS_ONLY, ep, true, {persist = true})
+      end
+
+      if supportedButtonValues_event then
+        device:emit_event_for_endpoint(ep, supportedButtonValues_event)
+      end
+      device:emit_event_for_endpoint(ep, capabilities.button.button.pushed({state_change = false}))
+    end
+  end
+end
+
+local function find_child(parent, ep_id)
+  return parent:get_child_by_parent_assigned_key(string.format("%d", ep_id))
+end
+
 local function initialize_switch(driver, device)
   local switch_eps = device:get_endpoints(clusters.OnOff.ID)
-  table.sort(switch_eps)
-  -- Since we do not support bindings at the moment, we only want to count On/Off
-  -- clusters that have been implemented as server. This can be removed when we have
-  -- support for bindings.
-  local num_server_eps = 0
-  local main_endpoint = find_default_endpoint(device)
-  for _, ep in ipairs(switch_eps) do
-    if device:supports_server_cluster(clusters.OnOff.ID, ep) then
-      num_server_eps = num_server_eps + 1
-      if ep ~= main_endpoint then -- don't create a child device that maps to the main endpoint
-        local name = string.format("%s %d", device.label, num_server_eps)
-        local child_profile = assign_child_profile(device, ep)
-        driver:try_create_device(
-          {
-            type = "EDGE_CHILD",
-            label = name,
-            profile = child_profile,
-            parent_device_id = device.id,
-            parent_assigned_child_key = string.format("%d", ep),
-            vendor_provided_label = name
-          }
-        )
+  local button_eps = device:get_endpoints(clusters.Switch.ID, {feature_bitmap=clusters.Switch.types.Feature.MOMENTARY_SWITCH})
+  local all_eps = {}
+
+  local new_profile = nil
+
+  local component_map = {}
+  local current_component_number = 2
+  local component_map_used = false
+  local parent_child_device = false
+
+  if #switch_eps > 0 or #button_eps > 0 then
+    for _,v in ipairs(switch_eps) do
+      table.insert(all_eps, v)
+    end
+    for _,v in ipairs(button_eps) do
+      table.insert(all_eps, v)
+    end
+    table.sort(all_eps)
+    -- Since we do not support bindings at the moment, we only want to count clusters
+    -- that have been implemented as server. This can be removed when we have
+    -- support for bindings.
+    local num_switch_server_eps = 0
+    local main_endpoint = find_default_endpoint(device)
+
+    for _, ep in ipairs(all_eps) do
+      if device:supports_server_cluster(clusters.Switch.ID, ep) then
+        -- Configure MCD for button endpoints
+        if tbl_contains(STATIC_PROFILE_SUPPORTED, #button_eps) then
```

Comment:
Can this be a clearer name? From just reading it I had no idea what it did. Sorry for all the naming convention stuff, I know you just ported a lot of this over

Reply:
Updated to `STATIC_BUTTON_PROFILE_SUPPORTED` to be more clear. This is just making sure that a profile exists for the given amount of button endpoints before attempting to index into the component to endpoint map.